### PR TITLE
harden sync sub committee selection

### PR DESCRIPTION
**Pull request by arnetheduck**
also turn it into an iterator

### CODE REVIEW

File: beacon_chain/consensus_object_pools/blockchain_dag.nim [hunk 1]
```diff
@@ -1081,7 +1078,7 @@ func getSubcommitteePositionsAux(
     return @[]
   let validatorPubKey = validatorKey.get().toPubKey
 
-  for pos, key in syncCommittee.syncSubcommittee(committeeIdx):
+  for pos, key in toSeq(syncCommittee.syncSubcommittee(committeeIdx)):
```
Comment (mratsim) [inline beacon_chain/consensus_object_pools/blockchain_dag.nim:1081]: Why not use `enumerate` or make the iterator return a tuple (pos, key)?

That does allocate a large sequence given that each ValidatorPubKey is 48 bytes.

**Review by mratsim — COMMENTED** [review]
What was the hardening?

File: beacon_chain/consensus_object_pools/blockchain_dag.nim (hunk 1 above)
Comment (arnetheduck) [reply beacon_chain/consensus_object_pools/blockchain_dag.nim:1081]: `enumerate` is 1.4+ and `pairs` requires a separate copy of the iterator (most of the time it's called without pos) - it was a seq before this change, so this is not a pessimization at least

**Comment by arnetheduck** [comment]
> What was the hardening?

out of bounds committee index leads to Defect which leads to crash

**Comment by arnetheduck** [comment]
(it happened in some test code we had)

**Review by tersec — APPROVED** [review]

File: beacon_chain/spec/forks.nim [hunk 2]
```diff
@@ -68,7 +68,7 @@ type
     of BeaconBlockFork.Merge:
       mergeBlock*:  merge.TrustedSignedBeaconBlock
 
-  ForkDigests* {.requiresInit.} = object
+  ForkDigests* = object
```
Comment (etan-status) [inline beacon_chain/spec/forks.nim:71]: Submarine fix.

**Pull request merged**